src/main.lua

```diff
@@ -877,31 +889,54 @@ Handlers.add(
 			)
 		end
 
-		local inputStatus, inputResult = pcall(checkAssertions)
-
-		if not inputStatus then
+		local shouldContinue = eventingPcall(msg.ioEvent, function(error)
 			ao.send({
 				Target = msg.From,
 				Tags = { Action = "Invalid-Decrease-Operator-Stake-Notice", Error = "Bad-Input" },
-				Data = tostring(inputResult),
+				Data = tostring(error),
 			})
+		end, checkAssertions)
+		if not shouldContinue then
 			return
 		end
-		local status, result =
-			pcall(gar.decreaseOperatorStake, msg.From, tonumber(msg.Tags.Quantity), msg.Timestamp, msg.Id)
-		if not status then
+
+		local quantity = tonumber(msg.Tags.Quantity)
+		msg.ioEvent:addField("Sender-Previous-Balance", balances[msg.From])
+
+		local shouldContinue2, gateway = eventingPcall(msg.ioEvent, function(error)
 			ao.send({
 				Target = msg.From,
 				Tags = { Action = "Invalid-Decrease-Operator-Stake-Notice", Error = "Invalid-Stake-Decrease" },
-				Data = tostring(result),
-			})
-		else
-			ao.send({
-				Target = msg.From,
-				Tags = { Action = "Decrease-Operator-Stake-Notice" },
-				Data = json.encode(result),
+				Data = tostring(error),
 			})
+		end, gar.decreaseOperatorStake, msg.From, quantity, msg.Timestamp, msg.Id)
+		if not shouldContinue2 then
+			return
 		end
+
+		msg.ioEvent:addField("Sender-New-Balance", balances[msg.From]) -- should be unchanged
+		if gateway ~= nil then
+			local previousStake = gateway.operatorStake + quantity
+			msg.ioEvent:addField("GW-New-Operator-Stake", gateway.operatorStake)
+			msg.ioEvent:addField("GW-Previous-Operator-Stake", previousStake)
+			msg.ioEvent:addField("GW-Vaults-Count", #(gateway.vaults or {}))
```

Comment:
see same comment as above

Reply:
`GW-Vaults-Count` is a holdover from Leave-Network where multiple kinds of vault data is returned. In fairness, they're all gateway vaults (as opposed to delegate vaults). This also distinguishes against ordinary vaults that might be associated with the wallet holder. I can remove the GW- prepend from the operator stake ones though.

Reply:
a5edd23